extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSFilter.java

```diff
@@ -323,4 +330,61 @@ static boolean substringMatch(String str, int pos, String substring, boolean req
             subPos++;
         }
     }
+
+    private void processPreFlightRequestedHeaders(HttpServerResponse response, String allowHeadersValue) {
+        if (isConfiguredWithWildcard(corsConfig.headers)) {
+            response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_HEADERS, allowHeadersValue);
+        } else {
+            Map<String, String> requestedHeaders;
+            String[] allowedParts = COMMA_SEPARATED_SPLIT_REGEX.split(allowHeadersValue);
+            requestedHeaders = new HashMap<>();
+            for (String requestedHeader : allowedParts) {
+                requestedHeaders.put(requestedHeader.toLowerCase(), requestedHeader);
+            }
+
+            List<String> corsConfigHeaders = corsConfig.headers.get();
+            StringBuilder allowedHeaders = new StringBuilder();
+            boolean isFirst = true;
+            for (String configHeader : corsConfigHeaders) {
+                String configHeaderLowerCase = configHeader.toLowerCase();
+                if (requestedHeaders.containsKey(configHeaderLowerCase)) {
+                    if (isFirst) {
+                        isFirst = false;
+                    } else {
+                        allowedHeaders.append(',');
+                    }
+                    allowedHeaders.append(requestedHeaders.get(configHeaderLowerCase));
+                }
+            }
+
+            response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_HEADERS, allowedHeaders.toString());
+        }
+    }
+
+    private void processPreFlightMethods(HttpServerResponse response, String allowMethodsValue) {
+        if (wildcardMethod) {
+            response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_METHODS, allowMethodsValue);
+        } else {
+            StringBuilder allowMethods = new StringBuilder();
```

Comment:
Tiny gripe: We can provide a good estimate of the size here.

Reply:
Hi @geoand, sorry, missed this comment, if the default is 16 then I guess it would be just right, ex, we have tests with 4 configured methods, `GET,POST,PUT,DELETE`, but these are 4 main methods, the users can just set a wildcard :-). So sometimes it can overflow by a few characters, but it is more likely than not 16 chars should be enough. Does it sound reasonable ?

Reply:
My point is that the we already know exactly what is going in to the builder, so we know the size.

In any case, it's a minor thing that can be addressed in a follow up

Reply:
Sure, I think I see the point; I might've seen it yesterday now that you have explained it but I hesitated as I was not too keen on checking the already configured methods length... In any case thanks for helping out with the performance advice :-)